validator/src/journal/candidate_block.rs

```diff
@@ -411,16 +411,6 @@ impl CandidateBlock {
                         "Batch {} is invalid, due to missing txn dependency",
                         header_signature
                     );
-                    bad_batches.push(batch.clone());
-                    pending_batches.clear();
-                    pending_batches.append(
-                        &mut self
-                            .pending_batches
-                            .clone()
```

Comment:
Clone here would filter bad batches out of the the cloned pending_batches.  There's a return statement soon after these three statements, thus it's redundant execution without useful result.

Reply:
An additional question could be made: is that return statement correct?

Reply:
I had this question, but added a comment in another PR here https://github.com/hyperledger/sawtooth-core/pull/1994 . These lines here seem to be solving the other issue, but due to lack of comments in the code I couldn't be very sure. If my understanding is correct then I will go ahead and clean this part to behave the way it's expected.